docker-compose.dev.yml

```diff
@@ -1,39 +1,98 @@
     build: "."
     # To make volume mounting (especially for new volume mounts like the DB)
     # easier, override the micromamba non-root user:
+    # TODO: remove?
     user: "root"
     volumes:
       - "./usaon_benefit_tool:/usaon-benefit-tool/usaon_benefit_tool:ro"
       - "./tasks:/usaon-benefit-tool/tasks:ro"
-      - "./_db:/db:rw"
     ports:
       - "5000:5000"
     environment:
+      ###############################################################
+      # DEV DANGER ZONE: These settings are INSECURE IN PRODUCTION. #
+      ###############################################################
+      # Enable the in-browser debugger:
+      FLASK_DEBUG: "True"
+      # In dev, we don't need to secure our sessions
+      FLASK_SECRET_KEY: "supersecret"
+
+      # Connect to the "db" service in this compose file!
+      USAON_BENEFIT_TOOL_DB_HOST: "db"
+      USAON_BENEFIT_TOOL_DB_PORT: "5432"
+      # Creds for talking to the DB:
+      USAON_BENEFIT_TOOL_DB_USER: &db-user "usaon-benefit-tool"
+      USAON_BENEFIT_TOOL_DB_PASSWORD: &db-pw "DevelopmentONLYPassword"
       # Disable login:
-      - "USAON_BENEFIT_TOOL_LOGIN_DISABLED=true"
-      # Use a local file DB instead of a remote one:
-      - "USAON_BENEFIT_TOOL_DB_SQLITE=true"
-        # Enable the in-browser debugger: 
-      - "FLASK_DEBUG=True"
-      # NOTE: This is temporary until issue #144 is resolved
+      USAON_BENEFIT_TOOL_LOGIN_DISABLED: "true"
+      # NOTE: Uncomment below if you want to test real SSO login behavior!
       # Creds for talking to Google:
-      - "USAON_BENEFIT_TOOL_GOOGLE_CLIENT_ID=${USAON_BENEFIT_TOOL_GOOGLE_CLIENT_ID:?USAON_BENEFIT_TOOL_GOOGLE_CLIENT_ID must be set}"
-      - "USAON_BENEFIT_TOOL_GOOGLE_CLIENT_SECRET=${USAON_BENEFIT_TOOL_GOOGLE_CLIENT_SECRET:?USAON_BENEFIT_TOOL_GOOGLE_CLIENT_SECRET must be set}"
-    
+      # USAON_BENEFIT_TOOL_GOOGLE_CLIENT_ID: "${USAON_BENEFIT_TOOL_GOOGLE_CLIENT_ID:?USAON_BENEFIT_TOOL_GOOGLE_CLIENT_ID must be set}"
+      # USAON_BENEFIT_TOOL_GOOGLE_CLIENT_SECRET: "${USAON_BENEFIT_TOOL_GOOGLE_CLIENT_SECRET:?USAON_BENEFIT_TOOL_GOOGLE_CLIENT_SECRET must be set}"
 
-      # Development DANGER ZONE: Do not use the below settings except for
-      # development. These settings are insecure in production.
-
-      # In dev, we don't need to secure our sessions
-      - "FLASK_SECRET_KEY=supersecret"
-      # Allow http in dev
+      # Allow http in dev; TODO: Do we need this? Why? Login is
+      # disabled in dev. Maybe as a fallback for actually testing
+      # login behavior?
       #     https://flask-dance.readthedocs.io/en/v0.8.0/quickstarts/google.html#code
-      - "OAUTHLIB_INSECURE_TRANSPORT=1"
+      # OAUTHLIB_INSECURE_TRANSPORT: "1"
+
+
+  # Normally, the database is managed independently of the app, but in dev, we
+  # want it to be easy to get started!
+  db:
+    <<: *common
+    depends_on:
+      db-volume-prep:
+        condition: "service_completed_successfully"
```

Comment:
See `#docker` Slack channel for more on what I learned about the Compose Specification!